doc/source/dev/toolchain.rst

```diff
@@ -124,7 +125,7 @@ Currently, SciPy wheels are being built as follows:
  Platform          Azure Base Image [5]_     Compilers                    Comment
 ================  ========================  ===========================  ==============================
 Linux (nightly)    ``ubuntu-20.04``          GCC 6.5                      See ``azure-pipelines.yml``
-Linux (release)    ``ubuntu-18.04``          GCC 7.5                      Built in separate repo [6]_
+Linux (release)    ``ubuntu-22.04``          GCC 8                        Built in separate repo [6]_
```

Comment:
this all needs to be re-written at some point, because we know do all the wheel build in `wheels.yml`, in this repo, with cibuildwheel, with more recent compilers, on a macOS-11 image, etc. Probably not something for this PR.